### PR TITLE
Migrate to ts: ErrorBoundary

### PR DESCRIPTION
**Pull request by marekdano**
### Description

This pull request converts [ErrorBoundary](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Error/ErrorBoundary.js) to typescript

### Changelog

- [ ] `CHANGELOG.md` is updated with user-facing content relevant to this PR.

### Affected components

- [ ] Server
- [x] Console
- [ ] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System
- [ ] Tests
- [ ] Other (list it)

### Related Issues

[#4314 console: migrate to TypeScript](https://github.com/hasura/graphql-engine/issues/4314)

#### Catalog upgrade

Does this PR change Hasura Catalog version?
- [x] No
- [ ] Yes
  - [ ] Updated docs with SQL for downgrading the catalog 

#### Metadata

Does this PR add a new Metadata feature?
- [x] No
- [ ] Yes
 
#### GraphQL
- [x] No new GraphQL schema is generated
- [ ] New GraphQL schema is being generated:
   - [ ] New types and typenames are correlated
   
   

#### Breaking changes

- [x] No Breaking changes
- [ ] There are breaking changes:

### CODE REVIEW

File: console/src/components/Error/ErrorBoundary.tsx [hunk 1]
```diff
@@ -11,15 +10,33 @@ import PageNotFound, { NotFoundError } from './PageNotFound';
...
-  initialState = {
+export type DefaultState = {
+  inconsistentObjects: Array<object>;
+  ongoingRequest: boolean;
+  allowedQueries: Array<object>;
+}
+
+export interface ErrorBoundaryProps {
+  metadata: DefaultState;
+  dispatch: (arg: object) => Promise<object>;
```
[the remark this answers is not in the document]
Comment (marekdano) [reply console/src/components/Error/ErrorBoundary.tsx:21]: That's a very good idea with `unknown`!

**Comment by marekdano** [comment]
@beerose I did changes as you suggested above and now getting `Unused state field: 'info'                             react/no-unused-state`. When I looked into `info`, I found that it's not used/passed to another component(s). Do you think we should keep it there and pass into `PageNotFound`or `RuntimeError` or remove it in the `ErrorBoundary`? Do we actually need it there?

**Comment by marekdano** [comment]
I've pushed the code where I added `info` as a new prop to `RuntimeError` component.

File: console/src/components/Error/ErrorBoundary.tsx [hunk 2]
```diff
@@ -11,25 +10,35 @@ import PageNotFound, { NotFoundError } from './PageNotFound';
...
 import { registerRunTimeError } from '../Main/Actions';
 
-class ErrorBoundary extends React.Component {
-  initialState = {
-    hasError: false,
-    info: null,
-    error: null,
-    type: '500',
-  };
+export type Metadata = {
```
Comment (beerose) [inline console/src/components/Error/ErrorBoundary.tsx:13]: For consistency
```suggestion
export interface Metadata {
```

File: console/src/components/Error/ErrorBoundary.tsx [hunk 3]
```diff
@@ -60,9 +69,20 @@ class ErrorBoundary extends React.Component {
     );
   }
 
+  initialState = {
```
Comment (beerose) [inline console/src/components/Error/ErrorBoundary.tsx:72]: Can you move it outside of the component and use `ErrorBoundaryState` to add a type to it, as it's not dependent on component props or state?

File: console/src/components/Error/ErrorBoundary.tsx [hunk 4]
```diff
@@ -60,9 +69,20 @@ class ErrorBoundary extends React.Component {
...
+    hasError: false,
+    info: null,
+    error: null,
+    type: '500',
+  };
+
+  resetState = () => {
+    this.setState({ ...this.initialState });
```
Comment (beerose) [inline console/src/components/Error/ErrorBoundary.tsx:80]: I know it's old code, but could you change it to the code below as there's no need to copy here:
```suggestion
    this.setState(this.initialState);
```

File: console/src/components/Error/ErrorBoundary.tsx (hunk 3 above)
Comment (marekdano) [reply console/src/components/Error/ErrorBoundary.tsx:72]: No, there are no dependencies. I can move it outside of the component.

**Comment by beerose** [comment]
Hi @marekdano, what's the status here? There are some small changes requested, after that it would be ready to merge 😉

**Comment by marekdano** [comment]
Sorry @beerose, I totally forgot about it. It's done now!

**Pull request merged**